src/client/trackers/page-attention/AttentionTime.js

```diff
@@ -0,0 +1,141 @@
+// Automatically stop the attention timer after this time
+const ATTENTION_INTERVAL = 15000;
+
+// These events re/start the attention timer
+const ATTENTION_EVENTS = [
+	'load',
+	'click',
+	'focus',
+	'scroll',
+	'mousemove',
+	'touchstart',
+	'touchend',
+	'touchcancel',
+	'touchleave'
+];
+
+// These events pause the attention timer
+const ATTENTION_LOST_EVENTS = ['blur'];
+
+// These events will trigger the exit callback
+const PAGE_EXIT_EVENTS = ['beforeunload', 'unload', 'pagehide'];
+
+const defaultOptions = {
+	onExit: () => {},
+	debug: false
+};
+
+export default class AttentionTime {
+	constructor (options) {
+		this.options = { ...defaultOptions, ...options };
+
+		this.totalAttentionTime = 0;
+		this.hasExited = false;
+
+		this.init();
+	}
+
+	init () {
+		ATTENTION_EVENTS.forEach((event) => {
+			window.addEventListener(event, this.startAttention.bind(this));
+		});
+
+		ATTENTION_LOST_EVENTS.forEach((event) => {
+			window.addEventListener(event, this.endAttention.bind(this));
+		});
+
+		PAGE_EXIT_EVENTS.forEach((event) => {
+			window.addEventListener(event, this.handleExit.bind(this));
+		});
+
+		document.addEventListener('visibilitychange', this.handleVisibilityChange.bind(this));
+
+		const videoPlayers = document.querySelectorAll('video');
+
+		videoPlayers.forEach((element) => {
```

Comment:
`NodeList.prototype.forEach` isn't supported in IE11

Reply:
Good point, thankfully it should get polyfilled: https://github.com/Financial-Times/dotcom-page-kit/blob/master/packages/dotcom-ui-polyfill-service/src/polyfillServiceURLs.ts#L23